### PR TITLE
MODINREACH-68 Add REST endpoint for local server credentials authenti…

### PR DESCRIPTION
**Pull request by andrew-khodyka**
## Purpose
Add REST endpoint that will be called by EDGE service to authenticate local server key/secret

## Approach
- Add authentication REST controller
- Add authentication service
- Delete redundant column from the local_server_credentials table
- Verify the implementation with unit & integration tests

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [x] Were any API paths or methods changed, added or removed?
  - [x] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by dmtkachenko** [comment]
What about security hotspot sonar reports, is it really critical? Can it be resolved somehow

File: src/main/resources/swagger.api/inn-reach.yaml [hunk 1]
```diff
@@ -134,6 +134,37 @@ paths:
       operationId: deleteLocation
       parameters:
         - $ref: '#/components/parameters/id'
+  /authentication:
```
Comment (dmtkachenko) [inline src/main/resources/swagger.api/inn-reach.yaml:306]: once we apply API interface generation from tags, your endpoint interface will be named incorrectly.
I suggest to add `tags` attribute right now (see sample in https://github.com/folio-org/mod-inn-reach/pull/20)

**Comment by dmtkachenko** [comment]
Overall looks good

**Comment by andrew-khodyka** [comment]
> What about security hotspot sonar reports, is it really critical? Can it be resolved somehow

No, it's not, not in our case

**Review by dmtkachenko — APPROVED** [review]

**Pull request merged**